odoo/addons/base/models/ir_model.py

```diff
@@ -937,6 +914,165 @@ def _add_manual_fields(self, model):
+        if not field_id:
+            _logger.error("No field_id found for %s", field)
+            return []
+
+        result = []
+        for sequence, selection in enumerate(field.selection, start=10):
+            result.append({
+                'field_id': field_id,
+                'value': str(selection[0]),  # value may be an int
+                'name': selection[1],
+                'sequence': sequence,
+                'field_name': field.name,
+            })
+        return result
+
+    def _reflect_model(self, model):
+        """ Reflect the given model's fields. """
+        module = self._context.get('module')
+        fields_data = self._existing_field_data(model._name)
+        to_insert = []
+        to_xmlids = []
+        for name, field in model._fields.items():
+            if field.type not in ('selection', 'reference') :
+                continue
+
+            old_vals = fields_data.get(name)
+            new_vals = self._reflect_field_params(field)
+            if old_vals is None or \
+                  old_vals != new_vals:
+                to_insert.extend(new_vals)
+            if module and (module == model._original_module or module in field._modules):
+                to_xmlids.extend(new_vals)
+
+        if to_insert:
+            rows = [(s['field_id'], s['value'], s['name'], s['sequence']) for s in to_insert]
+            field_selection_ids = []
+            query = """
+                INSERT INTO ir_model_fields_selection(field_id, value, name, sequence)
+                VALUES {}
+                ON CONFLICT (field_id, value) DO UPDATE
+                SET (field_id, value, name, sequence) = (EXCLUDED.field_id, EXCLUDED.value, EXCLUDED.name, EXCLUDED.sequence)
+                RETURNING id
+            """.format(", ".join("%s" for row in rows))
+            self.env.cr.execute(query, rows)
+            row_ids = self.env.cr.fetchall()
+            for index, row_id in enumerate(row_ids):
+                # keep id to create ir.model.data
+                to_insert[index]['id'] = row_id[0]
+
+        if to_xmlids:
+            # the field value may contains exotic chars like spaces
+            def sanitize(value):
+                return value.replace('.', '_')  \
+                            .replace(' ', '_')  \
+                            .lower()
+
+            # create or update their corresponding xml ids
+            pattern = '%s.field_selection__%s__%s__%s'
+            model_name = model._name.replace('.', '_')
+            self.env['ir.model.data']._update_xmlids([
+                dict(
+                    xml_id=pattern % (module, model_name, field_vals['field_name'], sanitize(field_vals['value'])),
+                    record=self.browse(field_vals['id']))
+                for field_vals in to_xmlids
+            ])
+
+    @api.multi
+    def write(self, vals):
+        if self.filtered(lambda s: s.field_id.state != 'manual'):
```

Comment:
This makes the condition more readable to me:
```suggestion
        if any(record.field_id.state != 'manual` for record in self):
```